File: java/src/main/java/de/gdata/vaas/exceptions/VaasAuthenticationException.java
package de.gdata.vaas.exceptions;

/**
 * The Vaas authentication failed.
 * Recommended actions:
 * * Check the clientId and clientSecret passed to ClientCredentialsGrantAuthenticator.
 * * Make sure the token has not expired.
 */
public class VaasAuthenticationException extends Exception {
    public VaasAuthenticationException(String message, Throwable cause) {
        super(message != null ? message : "Authentication error", cause);
    }

    public VaasAuthenticationException(String message) {
        this(message, null);
    }

    public VaasAuthenticationException() {
        this(null, null);
    }
}

File: php/src/vaas/Exceptions/VaasAuthenticationException.php
<?php

namespace VaasSdk\Exceptions;

use Exception;

/**
 * The Vaas authentication failed.
 * Recommended actions:
 * * Double-check your credentials in the authenticator object.
 * * Check if your authenticator connects to the correct token endpoint.
 * * Check if the token endpoint is reachable.
 * * If your problem persists contact G DATA.
 */
class VaasAuthenticationException extends Exception
{
    public function __construct(string $message = 'Authentication failed', int $code = 0, Exception $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}